### PR TITLE
Fix bug in fdb_to_revinfo compatibility clause for format version 0 to 1

### PR DESCRIPTION
**Pull request by jaydoane**
## Overview

This fixes a bug in a two `fdb_to_revinfo` compatibility clauses where `?CURR_REV_FORMAT` was not replaced with a `1` when it was recently bumped from `1` to `2`, making it impossible to read data in format version `0`.

## Testing recommendations

AFAIK, there are currently no tests for these compatibility clauses.

## Checklist

- [x] Code is written and works correctly
- [ ] Changes are covered by tests
- [ ] Any new configurable parameters are documented in `rel/overlay/etc/default.ini`
- [ ] A PR for documentation changes has been made in https://github.com/apache/couchdb-documentation

### CODE REVIEW

File: src/fabric/src/fabric2_fdb.erl [hunk 1]
```diff
@@ -1253,6 +1255,21 @@ fdb_to_revinfo(Key, {?CURR_REV_FORMAT, _, _, _, _, _} = Val) ->
         rev_size => RevSize
     };
 
+fdb_to_revinfo(Key, {?CURR_REV_FORMAT, _, _, _, _} = Val) ->
```
Comment (davisp) [inline src/fabric/src/fabric2_fdb.erl]: This looks wrong to me. There shouldn't be two versions of the current winner rev value.

**Comment by nickva** [comment]
I think the bug here is in the sizes PR a few commits back:

https://github.com/apache/couchdb/commit/eb1a09e114dafc55fa9511d477b9ada4350134eb#diff-32274bcb4785f432811085d2e08c3580

(And I did miss it during the review, sorry about that)

There are compatibility clauses there https://github.com/apache/couchdb/blob/prototype/fdb-layer/src/fabric/src/fabric2_fdb.erl#L1272 to upgrade from rev format `0 -> 1` and `1 -> 2`. However `0 -> 1` became broken because it uses the `?CURR_REV_FORMAT` macro which was bumped from `1` to `2` in that commit. So we are effectively doing a `0` -> `2` jump but with version `1` tuple format.

**Comment by jaydoane** [comment]
Good find @nickva! I backed out my previous commit, and just changed the two buggy compatibility clauses, and the code now seems to read my old data fine. Seems like maybe we should have some tests for these upgrade clauses?

**Comment by nickva** [comment]
@jaydoane if you feel up to a challenge :-) we have something similar to check reading in older local doc formats:

https://github.com/apache/couchdb/blob/prototype/fdb-layer/src/fabric/test/fabric2_doc_crud_tests.erl#L845-L893

If you start from version 0 that would run through version 1 and version 2 so you'd exercise all that code. If building those rev tuples from scratch is hard, maybe it's easier to create a few revisions via `fabric2_db:...` then read the tuples, shorten them and write them back (with plain old `erlfdb:set` as version 0). Then go through `fabric2_db` to read the revision and assert they were read correctly.

**Comment by jaydoane** [comment]
@nickva I opted for a simple unit test, which adds coverage to the formerly buggy code, and should help ensure the same type of bug is not re-introduced. I'm happy to pursue a more robust test if you think it's necessary, however.

**Comment by jaydoane** [comment]
@davisp are you ok with this solution?

**Pull request merged**